zerver/lib/actions.py

```diff
@@ -1915,6 +1916,27 @@ def do_change_avatar_source(user_profile, avatar_source, log=True):
                         person=payload),
                    active_user_ids(user_profile.realm))
 
+
+def do_change_icon_source(realm, icon_source, log=True):
+    # type: (Realm, Text, bool) -> None
+    realm.icon_source = icon_source
+    realm.save(update_fields=["icon_source"])
+
+    if log:
+        log_event({'type': 'realm_change_icon_source',
+                   'realm': realm.domain,
+                   'avatar_source': icon_source})
+    else:
+        payload = dict(
+            realm=realm.domain,
+            icon_url=realm_icon_url(realm)
+        )
```

Comment:
Since the realm is implied by the fact one is receiving the update, I think we just can simplify the event structure a lot.

e.g. 

`dict(type='realm_change_icon', url=realm_icon_url(realm))`